interfaces/cython/cantera/test/test_composite.py

```diff
@@ -22,9 +14,8 @@ class TestModels(utilities.CanteraTest):
     @classmethod
     def setUpClass(cls):
         utilities.CanteraTest.setUpClass()
-        cls.yml_file = pjoin(cls.test_data_dir, "thermo-models.yaml")
-        with open(cls.yml_file, 'rt', encoding="utf-8") as stream:
-            cls.yml = yaml.safe_load(stream)
+        cls.yml_file = cls.test_data_path / "thermo-models.yaml"
+        cls.yml = utilities.load_yaml(cls.yml_file)
```

Comment:
I'm wondering when it makes sense to turn `cls.test_data_dir` into a `Path`, I think we were waiting to drop 3.5 support because 3.6 was the first version to support `os.Pathlike` everywhere.

Reply:
I made the comment above before the changes were finished, so I'm not sure if it applies. Feel free to ignore/resolve if it's done now.

Reply:
Sorry for incrementally updating things: I left `cls.test_data_dir` as there would be a log of churn in `test_convert.py`. At this point, `CONTRIBUTING.md` assumes Python 3.6, so I believe turning everything into paths would be ok otherwise. Let me know whether to apply the changes ... which would add another 100 or so lines to thie PR.

Reply:
I say let's just go for the total change, no reason to leave old code lying around.